salt/react_apps.sls

```diff
@@ -25,39 +30,65 @@ include:
       - pkg: git
       - user: {{ entry.user }}_user_exists
 
-{{ appdir }}:
-  file.symlink:
-    - target: {{ directory }}
-    - makedirs: True
-    - runas: {{ entry.user }}
-    - require:
-      - git: {{ entry.git.url }}
-    # The symlink must be created before Apache starts.
-    - require_in:
-      - service: apache2
-
 {% if 'apache' in entry %}
 {{ apache(entry.git.target, entry.apache, context=context) }}
 {% endif %}
 
-{{ userdir }}-nvm-installer:
-  file.managed:
-    - name: {{ userdir }}/nvm-install.sh
-    - source: https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh
-    # curl -sS https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | shasum
-    - source_hash: e70e16f272f4ce4cc3e8d98e1e0ead449ab9ae5c
-    - user: {{ entry.user }}
-    - group: {{ entry.user }}
-    - mode: 755
+{{ directory }}/.env:
+  file.copy:
+    - source: {{ directory }}/.env.salt
 
-{{ userdir }}-nvm-install:
+{{ userdir }}-yarn-install:
   cmd.run:
-    - name: bash {{ userdir }}/nvm-install.sh
+    - name: yarn
     - runas: {{ entry.user }}
-    - cwd: {{ userdir }}
+    - cwd: {{ directory }}
     - require:
       - pkg: git
-    - onchanges:
-      - file: {{ userdir }}-nvm-installer
+
+{{ userdir }}-yarn-build:
+  cmd.run:
+    - name: yarn build
+    - env: {{ entry.env|yaml }}
+    - runas: {{ entry.user }}
+    - cwd: {{ directory }}
+    - require:
+      - pkg: git
+
+{% if enable_ver_txt %}
+
+{{ build_dir }}/ver.txt:
+  file.managed:
+    - contents: "branch: {{ entry.git.branch }} || commit_hash: {{ salt['cmd.shell']('cd '+ directory +' && git rev-parse --verify HEAD') }} || time: {{ tstamp }}"
+
+{% endif %}{# ver txt #}
+
+{{ directory }}/.htaccess:
+  file.managed:
+    - name: {{ build_dir }}/.htaccess
+    - source: {{ directory }}/.htaccess
+
+{{ build_dir }}/.htaccess:
+  file.replace:
+    - name: {{ build_dir }}/.htaccess
+    - pattern: 'PRERENDER_IO_TOKEN'
+    - repl: '{{ entry.prerender_token }}'
+
+{{ build_asset_dir }}-update:
+  cmd.run:
+    - name: rm -rf {{ build_asset_base }} && mkdir -p {{ build_asset_base }} && mv build {{ build_asset_base }}
+    - runas: {{ entry.user }}
+    - cwd: {{ directory }}
+
+{{ appdir }}:
+  file.symlink:
+    - target: {{ build_asset_base }}
+    - makedirs: True
+    - runas: {{ entry.user }}
+    - require:
+      - git: {{ entry.git.url }}
+    # The symlink must be created before Apache starts.
```

Comment:
Instead of depending on the order of the states in the file, you can just add a `require` so that this state depends on `{{ build_asset_dir }}-update`